cmake/externals/gsl.cmake

```diff
@@ -0,0 +1,18 @@
+if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.24.0")
+    FetchContent_Declare(
+        GSL
+        URL https://github.com/microsoft/GSL/archive/refs/tags/v4.0.0.zip
+        URL_HASH SHA1=cf368104cd22a87b4dd0c80228919bb2df3e2a14
+        FIND_PACKAGE_ARGS 4.0 NAMES Microsoft.GSL
+        DOWNLOAD_EXTRACT_TIMESTAMP TRUE
```

Comment:
This is the same setup used by ORT. Not sure if the FIND_PACKAGE_ARGS is really necessary but figured we do that in ORT for a good reason so should replicate here when the cmake version is new enough.